# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/GCVector.h
# Commit: 8e609f97c413
# Full Hash: 8e609f97c413a12f7042dfe80647bdd34f0400d8
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2019-11-01 16:13:11
# Regressor Bug: 1587096
# File Overlap Count: 1
# Description:
#   Bug 1587096 - Part 1: Add JS FinalizationGroupObject and related data structures r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49943
# ==============================================================================

diff -r f273ec2ec0ae -r 8e609f97c413 js/public/GCVector.h
--- a/js/public/GCVector.h	Fri Nov 01 10:37:09 2019 +0000
+++ b/js/public/GCVector.h	Fri Nov 01 10:37:25 2019 +0000
@@ -78,6 +78,17 @@
     return vector.append(std::forward<U>(item));
   }
 
+  void erase(T* it) { vector.erase(it); }
+  void erase(T* begin, T* end) { vector.erase(begin, end); }
+  template <typename Pred>
+  void eraseIf(Pred pred) {
+    vector.eraseIf(pred);
+  }
+  template <typename U>
+  void eraseIfEqual(const U& u) {
+    vector.eraseIfEqual(u);
+  }
+
   template <typename... Args>
   MOZ_MUST_USE bool emplaceBack(Args&&... args) {
     return vector.emplaceBack(std::forward<Args>(args)...);
@@ -273,6 +284,14 @@
   }
   void erase(T* aT) { vec().erase(aT); }
   void erase(T* aBegin, T* aEnd) { vec().erase(aBegin, aEnd); }
+  template <typename Pred>
+  void eraseIf(Pred pred) {
+    vec().eraseIf(pred);
+  }
+  template <typename U>
+  void eraseIfEqual(const U& u) {
+    vec().eraseIfEqual(u);
+  }
 };
 
 template <typename Wrapper, typename T, typename AllocPolicy>